src/pybind/mgr/volumes/fs/operations/versions/__init__.py

```diff
@@ -44,24 +48,43 @@ def _get_subvolume_version(self, version):
     def get_subvolume_object_max(self, fs, vol_spec, group, subvolname):
         return self._get_subvolume_version(self.max_version)(fs, vol_spec, group, subvolname)
 
+    def upgrade_to_v2_subvolume(self, subvolume):
+        # legacy mode subvolumes cannot be upgraded to v2
+        if subvolume.legacy_mode:
+            return
+
+        version = int(subvolume.metadata_mgr.get_global_option('version'))
+        if version >= SubvolumeV2.version():
+            return
+
+        v1_subvolume = self._get_subvolume_version(version)(subvolume.fs, subvolume.vol_spec, subvolume.group, subvolume.subvolname)
```

Comment:
Should attempt opening the v1 subvolume and handle not complete errors, before proceeding (in flight clones, or clone failed/cancelled state subvolumes should not be upgraded as yet).